### DISCUSSION TITLE
ECONNRESET Socket hang up

### DISCUSSION THREAD

**Discussion by flux0uz** [vercel/next.js#50894] [Help]
### Summary

For some time now, we've been receiving a lot "ECONNRESET socket hang up" errors in the Vercel & Axiom logs and some of our users are landing on page 500. This doesn't happen to everyone and it's not systematic. It happens on both SSG and SSR pages.

After hours and hours of searching, it's impossible to locate the source. We've checked that all the functions called in these pages have 'await' but we don't understand where this could be coming from.

We are on Next 13.4.3 using Page router.
Has anyone had the same problem or know where it might come from?

**Comment by mohammedsabi** [comment]
same issue

**Comment by DonovanCharpin** [comment]
I think I had the same issue you describe. We found our that our tcp_mem was increasing (memory leak) from nextjs.

 It seems that the bug was introduced when upgrading to v13.4.0. I could not find the source of the bug though, it might be linked to undici update but not sure.
 
 I did some tests and it seems that the bug is resolved in version 13.4.9!

**Reply by mdluo** [reply]
Yes, it was [next@13.3.5-canary.9](https://github.com/vercel/next.js/releases/tag/v13.3.5-canary.9). I had a comment here: https://github.com/vercel/next.js/issues/51605#issuecomment-1681004679
